doc/geom/SpanSet.rst

```diff
@@ -0,0 +1,181 @@
+given ``SpanSet`` is contiguous, and a method which will split an
+non-contiguous ``SpanSet`` into a standard vector (or list in python) of
+contiguous ``SpanSet`` objects. The following example demonstrates this behavior
+along side set operations.
+
+.. code-block:: python
+
+    from lsst.afw.geom import SpanSet, Stencil
+
+    # Construct some SpanSets
+    ss1 = SpanSet.fromShape(5)
+    ss2 = SpanSet.fromShape(5, offset=(6, 6))
+
+    # Create a union of the immutable sets
+    ss3 = ss1.union(ss2)
+
+    # Erode the combine SpanSet with a circular kernel and check if the result
+    # is contiguous
+    ssEroded = ss3.eroded(4, Stencil.CIRCLE)
+    assert(ssEroded.isContiguous() == False)
+
+    # Split the non contiguous SpanSet
+    spanSetSplitList = ssEroded.split()
+    assert(len(spanSetSplitList) == 2)
+
+
+SpanSets, Masks, Images, and Ndarray
+====================================
+Masks and ``SpanSet``\s can often be used in similar fashion to represent areas
+of interest in an image. As such ``SpanSet``\s provide many utility functions to
+make transporting this information between these two data types similar. As
+Masks are similar to ``SpanSet``\s they can even participate in ``SpanSet``
+mathematical operations. Examples for some of the ways masks and ``SpanSet``\s
+work together can be found in the example below.
+
+.. code-block:: python
+
+    from lsst.afw.geom import SpanSet
+    from lsst.afw.image import MaskU
+
+    # Create a mask to be populated
+    mask = MaskU(10, 10)
+
+    # Create a SpanSet which represents the pixels to be set in the mask, and
+    # set bit two
+    ss = SpanSet.fromShape(4, offset=(4,4))
+    ss.setMask(mask, 2)
+
+    # Intersect not (~) the SpanSet with the mask, the result should be a null
+    # SpanSet
+    ssIntersectNot = ss.intersectNot(mask)
+
+    # Convert the mask into a SpanSet and verify it evaluates equal to the
+    # origional
+    newSS = SpanSet.fromMask(mask)
+    assert(newSS == ss)
+
+As mentioned above the ``SpanSet`` class is used to encode sets of x, y
+locations on an image. These locations can be used to interact with
+additional images through a series of convince methods demonstrated in the
+subsequent example.
+
+.. code-block:: python
+
+    from lsst.afw.geom import SpanSet
+    from lsst.afw.image import ImageI
+
+    spanSet1 = SpanSet.fromShape(3, offset=(3, 3))
+    spanSet2 = SpanSet.fromShape(2, offset=(5, 5))
+
+    image1 = ImageI(10, 10)
+    image2 = ImageI(10, 10)
+
+    spanSet1.setImage(image1, 10)
+    spanSet2.setImage(image2, 15)
+
+    spanSet2.copyImage(image2, image1)
+
+    print(image1.getArray())
+
+A ``SpanSet`` can also be used to extract or insert values from/to ndarrays while
+expanding or reducing dimensionality. the ``flatten`` method extracts data from
+an array at locations defined by the ``SpanSet`` and returns (or inserts into)
+an ``ndarray`` with one less dimension. The ``unflatten`` method does the
+opposite.
```

Comment:
So if I have a (5 x 4 x 3 x 2) array. Which elements are selected and what dimensions will the array have that I get back?